File: src/RESTworld/RESTworld.Client.AspNetCore/DependencyInjection/ApiUrl.cs
namespace RESTworld.Client.AspNetCore.DependencyInjection
{
    /// <summary>
    /// An URL to a backend API that is used by the client to discover available APIs.
    /// </summary>
    public class ApiUrl
    {
        private string? _url;

        /// <summary>
        /// The name of the API.
        /// </summary>
        public string? Name { get; set; }

        /// <summary>
        /// The URL to the root of the API.
        /// </summary>
        public string? Url
        {
            get => _url;
            set => _url = (value?.EndsWith('/')).GetValueOrDefault() ? value : value + '/';
        }

        /// <summary>
        /// The version of the API.
        /// </summary>
        public int? Version { get; set; }
    }
}

File: src/Example/ExampleBlog/ExampleBlog.Client.Angular/ClientApp/src/app/ngx-restworld-client/models/api-url.ts
/**
 * The URL to a backend API that may be called.
 */
export interface ApiUrl {
  /**
   * The name of the API.
   */
  name: string;
  /**
   * The URL of the API.
   */
  url: string;
  /**
   * The version of the API.
   */
  version?: number;
}
